### PR TITLE
MM-16860 Support Guest Authentication via SAML

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Adds the SAMLSettings.GuestAttribute support to the config file.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-16860

### CODE REVIEW

**Comment by srkgupta** [comment]
Hi @sbishel 
The changes looks good. However I am unable to find any changes relevant to the following point mentioned in the JIRA Ticket:
`If this attribute is removed from your user in SAML and the user is still active, the user will be promoted to a member in Mattermost. `

Can you please let me know if I am missing something here?

**Comment by sbishel** [comment]
@srkgupta The user promotion/demotion functionality is in the enterprise project.  Here is the link to those changes.
https://github.com/mattermost/enterprise/pull/504
The server changes are simply updating the config file to support and validate the setting.  Let me know if you have further questions.

File: model/config.go [hunk 1]
```diff
@@ -2835,6 +2840,15 @@ func (ss *SamlSettings) isValid() *AppError {
 		if !(*ss.CanonicalAlgorithm == SAML_SETTINGS_CANONICAL_ALGORITHM_C14N || *ss.CanonicalAlgorithm == SAML_SETTINGS_CANONICAL_ALGORITHM_C14N11) {
 			return NewAppError("Config.IsValid", "model.config.is_valid.saml_canonical_algorithm.app_error", nil, "", http.StatusBadRequest)
 		}
+
+		if len(*ss.GuestAttribute) > 0 {
+			if !(strings.Contains(*ss.GuestAttribute, "=")) {
```
Comment (jespino) [inline model/config.go]: We need to decide it, but we can decide it later, is not needed to decide it before the merge.

Comment (sbishel) [reply model/config.go]: Yes, we still have an open issue in the Enterprise PR for this feature. If that changes, I will need to change this as well.

**Pull request merged**